sound/soc/sof/intel/hda-loader.c

```diff
@@ -131,6 +131,12 @@ static int cl_dsp_init(struct snd_sof_dev *sdev, const void *fwdata,
 		goto err;
 	}
 
+	/* set DONE bit to clear the reply IPC message */
+	snd_sof_dsp_update_bits_forced(sdev, HDA_DSP_BAR,
+				       chip->ipc_ack,
+				       chip->ipc_ack_mask,
+				       chip->ipc_ack_mask);
+
```

Comment:
Maybe the sequence is correct, but the commit message and comment could be improved:

1. what does 'clear the reply IPC message' mean? You only enable IPC interrupts in step 6.
2. There is nothing in this code which refers to a PURGE_IPC, so some background information is missing. There is a purge request

Looks like there are two forms of IPC and two types of DONE bits...

Reply:
> Maybe the sequence is correct, but the commit message and comment could be improved:
> 
> 1. what does 'clear the reply IPC message' mean? You only enable IPC interrupts in step 6.

There is no IPC interrupt at this point, but there do be IPC message here, those IPC registers are being set, without triggering an IPC IRQ. :)

> 2. There is nothing in this code which refers to a PURGE_IPC, so some background information is missing. There is a purge request

It is in some lines above, the step /* step 2: purge FW request */

> 
> Looks like there are two forms of IPC and two types of DONE bits...

Reply:
@keyonjie The point here is to clear the DONE bit from `/* step 4: wait for IPC DONE bit from ROM */` right?
If we do not clear it before we enable IPC IRQ, we will that that error message later.